presto-main/src/test/java/com/facebook/presto/operator/TestHashAggregationOperator.java

```diff
@@ -704,4 +806,26 @@ public void close()
             };
         }
     }
+
+    private static class TestMemoryPool
+            extends MemoryPool
```

Comment:
I don't think `MemoryPool` was written with extensibility in mind. The fact there is no `final` on it doesn't mean much, since we seldom use `final` anyway. And this is fragile: the `MemoryPool` class doesn't need to call the `getFreeBytes` you overrode for anything. It does now, but i consider it coincidental.

Instead, instantiate regular `MemoryPool` of desired size. Then, instead of changing free/max memory, just `reserve()` or `free()` some memory directly from the test.